base_layer/core/src/chain_storage/lmdb_db/lmdb_db.rs

```diff
@@ -261,7 +267,30 @@ impl LMDBDatabase {
                     height,
                     hash,
                     accumulated_difficulty,
+                    from,
                 } => {
+                    // for security we check that the best block does exist, and we check the previous value
+                    // we dont want to check this if the prev block has never been set, this means a empty hash of 32
+                    // bytes.
+                    if from != vec![0; 32] {
```

Comment:
I think we should check (before update):
1. the new best block header exists (does the header hash exist i.e. `lmdb_exists`)
2. the new block previous hash is set to the current best block hash (from metadata)
3. The height is one more than the current best block height (from metadata)

Reply:
We then do not rely on `from` being set correctly - we can remove `from`

Reply:
Ok 2 and 3 are too restrictive given what is needed for pruned mode - 1. is already included below